pkgs/tools/networking/openvpn3/default.nix

```diff
@@ -0,0 +1,115 @@
+{ lib
+, stdenv
+, fetchFromGitHub
+, asio
+, autoconf-archive
+, autoreconfHook
+, dbus-python
+, docutils
+, glib
+, gobject-introspection
+, gtest
+, jinja2
+, jsoncpp
+, libcap_ng
+, libnl
+, libuuid
+, lz4
+, openssl
+, pkg-config
+, protobuf
+, pygobject3
+, tinyxml-2
+, wrapGAppsHook
+, wrapPython
+}:
+
+let
+  # Original repository has three submodules: openvpn3-core, asio and
+  # googletests.It's a huge pain, so we fetch openvpn3-core by hand
+  # and use the existing derivation for googletests. asio just works,
+  # apparently.
+  openvpn3-core = fetchFromGitHub {
+    owner = "OpenVPN";
+    repo = "openvpn3";
+    rev = "7765540e581c48721752bcad0b3d74b8397b1f73";
+    sha256 = "sha256-v/suF/tWfuukQO1wFiHRzC7ZW+3Gh1tav6qj0uYUP4E=";
+  };
+in
+stdenv.mkDerivation rec {
+  pname = "openvpn3";
+  version = "17_beta";
+
+  src = fetchFromGitHub {
+    owner = "OpenVPN";
+    repo = "openvpn3-linux";
+    rev = "079e9da7f66f5cca59cf80ba8c548f0dedd433db";
+    sha256 = "sha256-ITSnC105YNYFW1M2bOASFemPZAh+HETIzX2ofABWTho=";
+  };
+  patches = [ ./dont-hardcode-gio.patch ]; # remove when v18_beta hits
+
+  postPatch = ''
+    substituteInPlace ./update-version-m4.sh --replace 'VERSION="$(git describe --always --tags)"' "VERSION=v${version}"
```

Comment:
> Please add a `leaveDotGit = true` to `fetchFromGitHub` if you're gonna do this.

no its not necessary to do that because this is hardcoding it instead of using git